pkg/volume/glusterfs/glusterfs.go

```diff
@@ -231,9 +231,41 @@ func (b *glusterfsMounter) SetUpAt(dir string, fsGroup *int64) error {
 	return err
 }
 
+// GlusterFsMountErrorHint performs some basic analysis
+// on the current mount error returned from the plugin
+// and will add a user hint or resolution tip for enhanced UXP
+// If no matches then original error is returned
+func GlusterFsMountErrorHint (inErr error) error {
+	if inErr == nil {
+		return nil
+	}
+	if dstrings.Contains(inErr.Error(), "lstat") && dstrings.Contains(inErr.Error(), "permission denied") {
+		return fmt.Errorf("%v\n\nAdditional Info: The pod is running, and the mount succeeded, however the mount is not accessible due to permissions.\nCheck the POSIX based permissions (owner, groups and others) on your mounted directory.\nIf needed, containers and pods can utilize and pass in a securityContext specifying runAsUser (uid/owner), or additional linux groups such as fsGroup (for block) or SupplementalGroups (for shared).\nWork with the storage adminstrator to ensure correct volume access.\n", inErr)
+	}
+	if dstrings.Contains(inErr.Error(), "endpoints") && dstrings.Contains(inErr.Error(), "not found") {
+		return fmt.Errorf("%v\n\nAdditional Info: Make sure the above endpoint exists, as they are needed for the node to communicate with the Gluster cluster.\nTo persist endpoints, they should be created as a service.\n", inErr)
+	}
+	if dstrings.Contains(inErr.Error(), "Connection timed out") || dstrings.Contains(inErr.Error(), "Transport endpoint is not connected") {
+		return fmt.Errorf("%v\n\nAdditional Info: Check and make sure the Gluster Server exists (ensure that correct IPAddress/Hostname was given in the endpoints) and is available/reachable.\n", inErr)
+	}
+	if dstrings.Contains(inErr.Error(), "failed to fetch PVC") || (dstrings.Contains(inErr.Error(), "persistentvolumeclaims") && (dstrings.Contains(inErr.Error(), "not found"))) {
+		return fmt.Errorf("%v\n\nAdditional Info: Check the pod spec to make sure the persistentVolumeClaim.name correctly matches the actual PVC name.\n", inErr)
+	}
+	if dstrings.Contains(inErr.Error(), "mount: unknown filesystem type") {
+		return fmt.Errorf("%v\n\nAdditional Info: Check and make sure the glusterfs-client package is installed (rpm -qa 'gluster*') on your nodes.\nIf not, install the client package on your nodes (i.e. yum install glusterfs-client -y).\nAfter installation to enable the gluster fuse client run 'modprobe fuse' on the node.\n", inErr)
```

Comment:
ditch the rpm hint.  Not everyone uses rpm, and I don't want things as brittle as package